skill-tree/src/main/java/com/RDS/skilltree/utils/UUIDValidationInterceptor.java

```diff
@@ -0,0 +1,41 @@
+package com.RDS.skilltree.utils;
+
+import com.RDS.skilltree.Exceptions.InvalidParameterException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+import org.springframework.web.servlet.HandlerInterceptor;
+import java.util.UUID;
+
+@Component
+public class UUIDValidationInterceptor implements HandlerInterceptor {
```

Comment:
middleware to perform UUID validation before passing it to the service

Reply:
can you please add these comments in the code itself

Reply:
Done